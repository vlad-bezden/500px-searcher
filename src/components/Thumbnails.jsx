import React from 'react'

import Thumbnail from './Thumbnail'

const Thumbnails = ({ photos, onClick }) => (
  <div>
    {photos.map(photo => (
      <div key={photo.id}
        style={{
          float: 'left',
          marginRight: '10px',
          marginBottom: '10px'
        }}>
        <a onClick={() => onClick(photo)} style={{ cursor: 'pointer' }}>
          <Thumbnail src={photo.image_url[0]} />
        </a>
      </div>
    ))}
  </div>
)

Thumbnails.propTypes = {
  photos: React.PropTypes.array.isRequired,
  onClick: React.PropTypes.func.isRequired
}

export default Thumbnails