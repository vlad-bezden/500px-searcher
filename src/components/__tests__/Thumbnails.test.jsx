import React from 'react'
import renderer from 'react-test-renderer'

import Thumbnails from '../Thumbnails'

import { MOCK_500PX_PHOTOS_RESPONSE } from '../../tests/mocks'

test('Thumbnails renders correctly', () => {
    const component = renderer.create(
        <Thumbnails photos={JSON.parse(MOCK_500PX_PHOTOS_RESPONSE).photos} />
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
})