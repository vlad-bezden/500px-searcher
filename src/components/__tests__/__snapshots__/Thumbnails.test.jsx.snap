// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnails renders correctly 1`] = `
<div>
  <div
    style={
      Object {
        "float": "left",
        "marginBottom": "10px",
        "marginRight": "10px",
      }
    }
  >
    <Thumbnail
      src="https://drscdn.500px.org/photo/63835479/w%3D280_h%3D280/67f4205dc8d9009c92d5fcb62abfd34b?v=0"
    />
  </div>
  <div
    style={
      Object {
        "float": "left",
        "marginBottom": "10px",
        "marginRight": "10px",
      }
    }
  >
    <Thumbnail
      src="https://drscdn.500px.org/photo/63730567/w%3D280_h%3D280/f8023f833a00ed5b0d5a78015dd6ce22?v=0"
    />
  </div>
  <div
    style={
      Object {
        "float": "left",
        "marginBottom": "10px",
        "marginRight": "10px",
      }
    }
  >
    <Thumbnail
      src="https://drscdn.500px.org/photo/76841387/w%3D280_h%3D280/0aeb7fa17244f346c9df2214a51d6f54?v=0"
    />
  </div>
  <div
    style={
      Object {
        "float": "left",
        "marginBottom": "10px",
        "marginRight": "10px",
      }
    }
  >
    <Thumbnail
      src="https://drscdn.500px.org/photo/99417191/w%3D280_h%3D280/14c607cd405d06f3bdd35f0d771f26e1?v=9"
    />
  </div>
  <div
    style={
      Object {
        "float": "left",
        "marginBottom": "10px",
        "marginRight": "10px",
      }
    }
  >
    <Thumbnail
      src="https://drscdn.500px.org/photo/69253929/w%3D280_h%3D280/4a118b07d2e92dc9ea740981581cf165?v=0"
    />
  </div>
  <div
    style={
      Object {
        "float": "left",
        "marginBottom": "10px",
        "marginRight": "10px",
      }
    }
  >
    <Thumbnail
      src="https://drscdn.500px.org/photo/96897331/w%3D280_h%3D280/ba41e8db5ed84074c078ea059ce4ae9c?v=11"
    />
  </div>
  <div
    style={
      Object {
        "float": "left",
        "marginBottom": "10px",
        "marginRight": "10px",
      }
    }
  >
    <Thumbnail
      src="https://drscdn.500px.org/photo/63425207/w%3D280_h%3D280/a26497a76099ecc12e1fe9e64fd653d3?v=0"
    />
  </div>
  <div
    style={
      Object {
        "float": "left",
        "marginBottom": "10px",
        "marginRight": "10px",
      }
    }
  >
    <Thumbnail
      src="https://drscdn.500px.org/photo/123195029/w%3D280_h%3D280/5c2ae602eed0a7a0cc88490ff75b1d34?v=6"
    />
  </div>
  <div
    style={
      Object {
        "float": "left",
        "marginBottom": "10px",
        "marginRight": "10px",
      }
    }
  >
    <Thumbnail
      src="https://drscdn.500px.org/photo/76796395/w%3D280_h%3D280/b85de1c7d1005433d40a14120696254a?v=0"
    />
  </div>
  <div
    style={
      Object {
        "float": "left",
        "marginBottom": "10px",
        "marginRight": "10px",
      }
    }
  >
    <Thumbnail
      src="https://drscdn.500px.org/photo/156174679/w%3D280_h%3D280/f91cde182e51aba534280936f5705aa0?v=3"
    />
  </div>
  <div
    style={
      Object {
        "float": "left",
        "marginBottom": "10px",
        "marginRight": "10px",
      }
    }
  >
    <Thumbnail
      src="https://drscdn.500px.org/photo/83674487/w%3D280_h%3D280/a36eef466e19b16c71660f203fc8c3f8?v=9"
    />
  </div>
  <div
    style={
      Object {
        "float": "left",
        "marginBottom": "10px",
        "marginRight": "10px",
      }
    }
  >
    <Thumbnail
      src="https://drscdn.500px.org/photo/63956645/w%3D280_h%3D280/b51bc06655225c876022013cdf8c94aa?v=0"
    />
  </div>
  <div
    style={
      Object {
        "float": "left",
        "marginBottom": "10px",
        "marginRight": "10px",
      }
    }
  >
    <Thumbnail
      src="https://drscdn.500px.org/photo/63802775/w%3D280_h%3D280/dc4fe9c7f7992d8aca5c451226d38341?v=0"
    />
  </div>
  <div
    style={
      Object {
        "float": "left",
        "marginBottom": "10px",
        "marginRight": "10px",
      }
    }
  >
    <Thumbnail
      src="https://drscdn.500px.org/photo/69176397/w%3D280_h%3D280/280e1124292f51ffc1859a60d98ad325?v=0"
    />
  </div>
  <div
    style={
      Object {
        "float": "left",
        "marginBottom": "10px",
        "marginRight": "10px",
      }
    }
  >
    <Thumbnail
      src="https://drscdn.500px.org/photo/72624433/w%3D280_h%3D280/21d7e37ecbe6f6d5309c746c0c6f4843?v=0"
    />
  </div>
  <div
    style={
      Object {
        "float": "left",
        "marginBottom": "10px",
        "marginRight": "10px",
      }
    }
  >
    <Thumbnail
      src="https://drscdn.500px.org/photo/87111511/w%3D280_h%3D280/de03f115b4f98454f3a04bf176cae807?v=5"
    />
  </div>
  <div
    style={
      Object {
        "float": "left",
        "marginBottom": "10px",
        "marginRight": "10px",
      }
    }
  >
    <Thumbnail
      src="https://drscdn.500px.org/photo/72887905/w%3D280_h%3D280/392182cd121ae0b4d596c691f1af1db7?v=0"
    />
  </div>
  <div
    style={
      Object {
        "float": "left",
        "marginBottom": "10px",
        "marginRight": "10px",
      }
    }
  >
    <Thumbnail
      src="https://drscdn.500px.org/photo/79253787/w%3D280_h%3D280/8e4d591ddfc79de4a796788eb060443a?v=0"
    />
  </div>
  <div
    style={
      Object {
        "float": "left",
        "marginBottom": "10px",
        "marginRight": "10px",
      }
    }
  >
    <Thumbnail
      src="https://drscdn.500px.org/photo/64324463/w%3D280_h%3D280/e34bc9f5a3fea96cd2eb6f4e7190502a?v=0"
    />
  </div>
  <div
    style={
      Object {
        "float": "left",
        "marginBottom": "10px",
        "marginRight": "10px",
      }
    }
  >
    <Thumbnail
      src="https://drscdn.500px.org/photo/105828979/w%3D280_h%3D280/c08b16e75062f340877c2f053d89188c?v=6"
    />
  </div>
</div>
`;
